service/topology_coordinator.cc

```diff
@@ -1664,6 +1667,14 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
         }
 
         if (drain) {
+            if (has_nodes_to_drain) {
+                // Prevent jumping to write_both_read_old with un-drained tablets.
+                // This can happen when all candidate nodes are down.
+                rtlogger.warn("Tablet draining stalled: No tablets migrating but there are nodes to drain");
+                release_guard(std::move(guard));
+                co_await sleep(3s); // Throttle the above logging
```

Comment:
If it's only about throttling the log message, why not use seastar::logger::rate_limit?

Reply:
Each cycle logs from many places, not just this one.

Reply:
Why 3s? Can't that be too much, or too little in certain conditions?

Reply:
If we're here, it's unlikely the situation will resolve itself soon, because either we have a bug in the scheduler which will require manual intervention, or the nodes required to make progress went down, in which case bringing them up requires time scale larger than 3s.

There is no particular reason for 3s other than it being sufficiently large to not flood the logs excessively in the recovery time scale